06-hot-reloading-with-directives/README.md

```diff
@@ -0,0 +1,99 @@
+# Example 6 – Hot reloading with stitching directives
+
+This example demonstrates the use of stitching directives to specify type merging configuration. Moving service-specific configuration out of the gateway facilitates service setup automation, facilitating hot-reloading.
+
+The `@graphql-tools/stitching-directives` package provides importable directives that can be used to annotate types and fields within subschemas, a validator to ensure the directives are used appropriately, and a configuration transformer that can be used on the gateway to convert the subschema directives into explicit configuration setting.
+
+The gateway server can refresh the schema in response to "push" input of service changes via mutations as well as "pull" input of service health via subschema SDL polling.
+
+Note: the service setup in this example is based on the [official demonstration repository](https://github.com/apollographql/federation-demo) for
+[Apollo Federation](https://www.apollographql.com/docs/federation/).
+
+**This example demonstrates:**
+
+- Adding remote schemas, with typedefs and custom directives exposed via a custom root field.
+- Use of the @key, @computed and @merge directives to specify type merging configuration.
+- Use of a custom executor that times out a request after a pre-specified limit.
+- Addition of custom queries/mutations on the gateway for listing/modifying the configured services.
+- Hot reloading of the gateway schema based on "push" input of service changes and "pull" input of service health.
+
+## Setup
+
+```shell
+cd 06-hot-reloading-with-directives.
+
+yarn install
+start-service-accounts
+```
+
+Then, in a separate terminal tab:
+
+```shell
+start-service-inventory
+```
+
+In a third terminal tab:
+
+```shell
+start-service-products
+```
+
+In a fourth terminal tab:
+
+```shell
+start-service-reviews
```

Comment:
Any reason this isn't just two terminal windows? The `concurrently` package runs services together. I stole this from Frederation demo, but you run `start-services` to run all subservices in the same tab together, and then run `start-gateway` to run it on its own.

Reply:
This is so for the second part of the demonstration, you can easily halt whichever services you want

Reply:
Gotcha. Suggested below... the four-part cluster in SDL demo could use concurrently, while the smaller two-part cluster for the hot loading demo could use one-offs, and call that out that you can shut down servers in the readme.